lib/utils/testem-events.js

```diff
@@ -152,31 +161,56 @@ class TestemEvents {
     };
   }
 
+  /**
+   * Creates a file with targetJsonObject
+   *
+   * @param {*} fileName
+   * @param {*} targetJsonObject
+   * @param {*} option
+   */
+  _writeJsonToFile(fileName, targetJsonObject, option = {}) {
+    try {
+      fs.writeJsonSync(fileName, targetJsonObject, option);
+    } catch (err) {
+      err.file =  err.file || fileName;
+      throw err;
+    }
+  }
+
+  /**
+   * Returns a sorted module metadata from statemanager.
+   */
+  _getSortedModuleMetaData(moduleMetadata) {
+    return moduleMetadata.sort((a, b) => { return b.runtime - a.runtime; });
+  }
+
   /**
    * Keep track of the number of browsers that completed executing its tests.
    * When all browsers complete, write test-execution.json to disk and clean up the stateManager
    *
    * @param {number} browserCount
    * @param {Object} ui
    * @param {boolean} loadBalance
-   * @param {string} fileName
+   * @param {string} currentDate
    * @param {string} writeExecutionFile
    */
-  completedBrowsersHandler(browserCount, ui, loadBalance, fileName, writeExecutionFile) {
+  completedBrowsersHandler(browserCount, ui, loadBalance, currentDate, writeExecutionFile) {
     let browsersCompleted = false;
 
     this.stateManager.incrementCompletedBrowsers();
     if (this.stateManager.getCompletedBrowser() === browserCount) {
+      const moduleDetailFileName = path.join(this.root, `module-run-details-${currentDate}.json`);
```

Comment:
Don't see a flag here for `module-run-details-.json`
Is the plan to write to disk for every run?

Reply:
I thought that the module metatdata file will be useful for every runs, but open to any thoughts :)

Reply:
I had the following scenario in mind:
- if the test suite is pretty well optimize, what would the user gain from this file? They may want to say I no longer need this file but would have no way to turn it off.

Side note: I'm sure this will come in future revision, documentation on what the file means and how it can be used.

Reply:
I agree.. If we are making to have module metadata file as optional I'm thinking shouldn't it be an optional for every cases whether or not it's using load-balance? Even without using load-balance I think the metadata can be somewhat useful for many cases..

Reply:
`Even without using load-balance I think the metadata can be somewhat useful for many cases.` Agreed! Although we don't have any metadata info for non-load balance runs atm.

Reply:
Glad you agreed! I think we still can use QUnit callbacks to make metadata available for both non-load balance and load balance.